packages/beacon-node/src/sync/backfill/backfill.ts

```diff
@@ -744,7 +744,7 @@ export class BackfillSync extends (EventEmitter as {new (): BackfillSyncEmitter}
 
   private async syncBlockByRoot(peer: PeerId, anchorBlockRoot: Root): Promise<void> {
     const [anchorBlock] = await this.network.reqResp.beaconBlocksByRoot(peer, [anchorBlockRoot]);
-    if (anchorBlock == null) throw new Error("InvalidBlockSyncedFromPeer");
+    if (anchorBlock === null) throw new Error("InvalidBlockSyncedFromPeer");
```

Comment:
type of `anchorBlock` is `allForks.SignedBeaconBlock[]` the compiler enforces this is not null, so this check is actually frivolous.